qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/parser/template/sections/BaseWhenSection.java

```diff
@@ -53,4 +85,26 @@ public ParametersInfo getParametersInfo() {
 	public List<SectionKind> getBlockLabels() {
 		return List.of(SectionKind.IS, SectionKind.CASE, SectionKind.ELSE);
 	}
+
+	private static boolean isSingleParamOperator(String partName) {
+		return singleParamOperators.containsKey(partName);
+	}
+
+	private static boolean isMultiParamOperator(String partName) {
+		return MultiParamOperators.containsKey(partName);
+	}
+
+	public static boolean shouldCompleteWhenSection(Section section) {

```

Comment:
I think you should replace the method with (not sure that it is correct but it is just to give you some direction):

```java
public static boolean shouldCompleteWhenSection(Section section) {
	int paramCount = section.getParameters().size();
	if (paramCount  == 0) {
		return true;
   	}
   	CaseOperator operator = caseOperators.get(section.getParameterAtIndex(0).getName());
   	if (operator == null) {
            if (paramCount == 1) {
            		// There is only parameter and it is not a operator
			return false;
            }
            return true;
        }
	if (paramCount == 2 && !operator.isMulti()) {
		// first parameter is operator that only allows single value, second is the
		// existing Enum
		return false;
	} 
	return true;
}
```